### DISCUSSION TITLE
Possible to edit the hover CSS for the button?

### DISCUSSION THREAD

**Discussion by sjnair** [dnplus/streamlit-oauth#46] [General]
This is a great tool, thanks for creating it!  I can't believe how messy authorization is, and this library makes it much easier.

The button seems to go red on hover by default.  It looks like that's defined in style.css?

```
button:hover {
  color: white;
  width:;
  /* box-shadow: 0 0 20px rgba(104, 85, 224, 0.6); */
  background-color: var(--primary-color);
}
```

Is it possible to define these colors in the Streamlit code, for instance here:
```
result = oauth2.authorize_button(
        name="Sign-in with Google",
        icon="https://www.google.com.tw/favicon.ico",
        redirect_uri="http://localhost:8501/",
        scope="openid email profile",
        key="google",
        extras_params={"prompt": "consent", "access_type": "offline"},
        use_container_width=False,
        pkce='S256',
    )
```
Or using this library means the button will always be red on hover?

I tried using [theming](https://docs.streamlit.io/develop/concepts/configuration/theming) via the config.toml file but it didn't seem to affect the hover color.

**Comment by dnplus** [comment]
Thanks for sharing

It's possible to add style in code, but obviously there are so many options can be added

Suggest use theming, works for me with theming method
![CleanShot 2024-07-08 at 09 36 18](https://github.com/dnplus/streamlit-oauth/assets/2977735/0fec48eb-5faa-47fe-a2b9-4a36e7fc530c)

**Reply by sjnair** [reply]
Thanks @dnplus, this seems to be due to a separate issue because regardless of anything I put in my config.toml file, I am not seeing the changes when I run my app.  So, nothing to do with streamlit-oauth!  Once I get that working, I will use the theming to change this.

**Comment by fabridb** [comment]
What if theming is not an option? How can i reach the button with the css? I set key="btgoogle" but .st-key-btgoogle is not giving the expected result. 
.st-key-btgoogle button{} doesn't work.

**Reply by dnplus** [reply]
I believe you need to do some tricks with css

but I don't have much experience with

**Comment by fabridb** [comment]
There is no way to reach the button i think.
What if authorize_button has one more parameter: array for CSS Script.
Anyway, thank you for your attention.


Em qui., 15 de mai. de 2025 às 23:48, Dylan Lu ***@***.***>
escreveu:
